src/data/autogenerated-forms-configurator/DataStoreConfigurationRepository.ts

```diff
@@ -46,11 +47,127 @@ export class Dhis2DataStoreConfigurationRepository implements DataStoreConfigura
             }));
     }
 
-    async getConstants(): Promise<string[]> {
-        const { constants } = await this.api
-            .get<{ constants: { code: string }[] }>("/constants?fields=id,code")
+    private async getMetadata() {
+        const { objects: dataSets } = await this.api.models.dataSets
+            .get({
+                fields: dataSetFields,
+                filter: {
+                    name: {
+                        ilike: "NHWA",
+                        like: "MAL",
+                    },
+                },
+                paging: false,
+                rootJunction: "OR",
+            })
             .getData();
 
+        return dataSets;
+    }
+
+    async getSelectedDataSet(dataSets: DataSet[], dataSetCode: Maybe<Code>): Promise<AutogenConfigSchema> {
```

Comment:
Having a method `getSelectedDataSet` inside `DataStoreConfigurationRepository` could be confusing. Normally a repository should work with an specific Entity.

My proposal would be:

This whole method should be in the `GetSelectedDataSetUseCase` file instead here.

```ts
export class GetSelectedDataSetUseCase {
    constructor(private constantsRepository: ConstantRepository) {}

    execute(dataSets: DataSet[], dataSetCode: Maybe<Code>) Promise<AutogenConfigSchema> {
        const dataSet = _.find(dataSets, { code: dataSetCode });
        if (!dataSetCode || !dataSet)
            return { sections: [], categoryComboCodes: [], dataElements: [], constantCodes: [] };

        // we need a get method in the constant repository
        const constantsCodes = await this.constantRepository.get();
        const dataElements = this.getDataElements(dataSet);
        const sections = this.getSections(dataSet);
        const categoryComboCodes = this.getCategoryComboCodes(dataSet);

        return {
            categoryComboCodes: categoryComboCodes,
            constantCodes: constantCodes,
            dataElements: dataElements,
            sections: sections,
        };
    }
}
```

and maybe changing the filename could be `GetAutogenConfigSchemaUseCase.ts` since that's the object we're building.

Does it make sense?